src/docfx/lib/JsonUtility.cs

```diff
@@ -135,6 +140,25 @@ public static (List<Error>, JToken) ValidateNullValue(this JToken token, JTokenS
             return (errors, token);
         }
 
+        public static List<Error> ValidateSchema(this JToken token, Type type, JTokenSourceMap mappings = null)
+        {
+            // TODO: Get the license somewhere
+            var registerErrors = RegisterLicense(string.Empty);
+            if (registerErrors.Any())
+                return registerErrors;
+
+            var generator = new JSchemaGenerator() { ContractResolver = new CamelCasePropertyNamesContractResolver() };
+            var schema = generator.Generate(type, true);
+            var schemaString = schema.ToString();
+            if (!token.IsValid(schema, out IList<ValidationError> errors))
+            {
+                var error = errors.First();
```

Comment:
>errors [](start = 28, length = 6)

output all errors?

Reply:
I thought we could just throw the first error to use to fix for further information. You are right, we should output all the errors for later usage.

---
In reply to: [201662860](https://github.com/dotnet/docfx/pull/3052#discussion_r201662860) [](ancestors = 201662860)